engine/consensus/dkg/reactor_engine.go

```diff
@@ -305,19 +310,24 @@ func (e *ReactorEngine) handleEpochCommittedPhaseStarted(currentEpochCounter uin
 			Str("canonical_beacon_pub_key", nextDKGPubKey.String()).
 			Msg("checking beacon key consistency: locally computed beacon public key does not match beacon public key for next epoch")
 		err := e.dkgState.SetDKGEndState(nextEpochCounter, flow.DKGEndStateInconsistentKey)
-		if err != nil {
+		if errors.Is(err, storage.ErrAlreadyExists) {
+			e.log.Warn().Err(err).Msg("failed to set dkg end state - already set")
+		} else if err != nil {
 			log.Fatal().Err(err).Msg("failed to set dkg end state")
 		}
 		return
 	}
 
 	err = e.dkgState.SetDKGEndState(nextEpochCounter, flow.DKGEndStateSuccess)
-	if err != nil {
+	if errors.Is(err, storage.ErrAlreadyExists) {
+		e.log.Warn().Err(err).Msg("failed to set dkg end state - already set")
```

Comment:
can this happen during normal operations? Isn't it a symptom of bug?

Reply:
Good point - with at-least-once delivery we can possibly get here during normal operations, but that isn't possible any more.

Reply:
https://github.com/onflow/flow-go/pull/2994/commits/0170a66c5d8d82ebc0b1fdae67c96fe1abb18226